From f0d25a37367e2a96cb14bd39dcf32c8b619d866a Mon Sep 17 00:00:00 2001
From: David Howells <dhowells@redhat.com>
Date: Tue, 25 Nov 2014 16:34:26 -0500
Subject: [fs] namei: trivial fix to vfs_rename_dir comment

Message-id: <20141125163426.18115.70578.stgit@warthog.procyon.org.uk>
Patchwork-id: 100921
O-Subject: [PATCH 21/47] namei: trivial fix to vfs_rename_dir comment
Bugzilla: 985875
RH-Acked-by: Steve Best <sbest@redhat.com>

From: J. Bruce Fields <bfields@redhat.com>

Looks like the directory loop check is actually done in renameat?
Whatever, leave this out rather than trying to keep it up to date with
the code.

Signed-off-by: J. Bruce Fields <bfields@redhat.com>
Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Original-commit: d03b29a271eb1d6de5af0f46cf0e7487e9e9284b
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/namei.c b/fs/namei.c
index b1e1e40..f801f1b 100644
--- a/fs/namei.c
+++ b/fs/namei.c
@@ -3959,7 +3959,7 @@ SYSCALL_DEFINE2(link, const char __user *, oldname, const char __user *, newname
  * The worst of all namespace operations - renaming directory. "Perverted"
  * doesn't even start to describe it. Somebody in UCB had a heck of a trip...
  * Problems:
- * a) we can get into loop creation. Check is done in is_subdir().
+ * a) we can get into loop creation.
  * b) race potential - two innocent renames can create a loop together.
  *    That's where 4.4 screws up. Current fix: serialization on
  *    sb->s_vfs_rename_mutex. We might be more accurate, but that's another
-- 
1.7.1